### ISSUE TITLE
Delete posting on the Wall returns you to /network

### ISSUE THREAD

**Issue by tobiasd** [friendica/friendica#6006]
### Expected behavior

When deleting a posting from your wall (`/profile/name`) you are returned to your wall again.

### Actual behavior

When deleting a posting from the wall, the user is returned to the network stream (`/network`)

### Friendica version you encountered the problem

2018.12-dev

**Comment by nupplaphil** [comment]
I'm afraid, it sounds like a bug because of #5907

**Comment by MrPetovan** [comment]
Isn't because of #5861?

**Comment by JonnyTischbein** [comment]
> Isn't because of #5861?

It is! https://github.com/friendica/friendica/pull/5861#discussion_r223811563

> The item has been successfully deleted, we should redirect to the /network page since the item doesn't exist anymore.

**Comment by MrPetovan** [comment]
This sounds lovely.

**Comment by JonnyTischbein** [comment]
I'm a bit confused why this issue occurses because most of the time it should be an AJAX call, isn't it ? Or the other way around: I implemented the redirect via an additional parameter but it is not executed because of (my changes are in `drop_item()`:

/mod/item.php 878-890

    function item_content(App $a)
    [...]
	if (($a->argc >= 3) && ($a->argv[1] === 'drop') && intval($a->argv[2])) {
		if ($a->isAjax()) {
			$o = Item::deleteForUser(['id' => $a->argv[2]], local_user());
		} else {
			$o = drop_item($a->argv[2], $a->argv[3]);
		}

		if ($a->isAjax()) {
			// ajax return: [<item id>, 0 (no perm) | <owner id>]
			echo json_encode([intval($a->argv[2]), intval($o)]);
			killme();
		}
	}

**Comment by MrPetovan** [comment]
(GitHub-foo: To show an extract of an existing file, you can go to the file (like https://github.com/friendica/friendica/blob/develop/mod/item.php), click on the number of the first line you want to show (here 878), then shift-click on the number of the last line you want to show (here 890), and then click on the "..." button and click on "Copy permalink". Once you paste it in your comments, it shows like this:

https://github.com/friendica/friendica/blob/884ab0d71d983bc6d452284a828126cad866562c/mod/item.php#L878-L890

**Comment by MrPetovan** [comment]
I'm a little bit confused by your comment, I feel like it's missing words after "but in case of `/display/123`". This makes "Or the other way around" even more confusing because I didn't think I read about the first way.

**Comment by JonnyTischbein** [comment]
Sorry!
I discovered another issue
1. Go to the `/display/[guid]`page of your post
2. Delete it via the dropdown menu
3. The post fades out but you stay on the `display` page

I guess this happens because of the referenced code lines. The redirection happens in `drop_item()` which is only called if not using AJAX (I don't know when this is the case).
If you call directly `/item/drop/id` (or in my testing branch `/item/drop/id/return_url`) it works fine.

I hope I could clear up my last post :crossed_fingers:

**Comment by tobiasd** [comment]
I'm using vier.

**Comment by MrPetovan** [comment]
@JonnyTischbein Yes, you're right, the deletion on `/display` works but leaves an empty screen. I'm not sure it is that big of a problem, at least it leaves the user with the choice of closing the tab of navigating elsewhere.

**Comment by MrPetovan** [comment]
I'm not using vier, can someone test again after #6056?

**Comment by JonnyTischbein** [comment]
I tested: Deletion a post in vier and quattro from
- `/network`
- `/profile/[user]`
- `/display/[guid]`
Also possible to test from
- `/community` 

It would be great if somebody could confirm this :)

**Comment by tobiasd** [comment]
Deleted a posting from my wall (using vier) and was returned to the wall again. As @JonnyTischbein tested this as well I guess the issue is solved :-)
